pkg/crc/machine/start.go

```diff
@@ -126,156 +125,6 @@ func restartHost(ctx context.Context, vm *virtualMachine, sshRunner *crcssh.Runn
 	return nil
 }
 
-func ocpGetPVShiftSize(diskSize int, pvSize int) int {
-	defaultPvSize := constants.GetDefaultPersistentVolumeSize(crcPreset.OpenShift)
-	if pvSize > defaultPvSize {
-		return (diskSize - constants.DefaultDiskSize) - (pvSize - defaultPvSize)
-	}
-	return diskSize - constants.DefaultDiskSize
-}
-
-func growRootFileSystem(ctx context.Context, startConfig types.StartConfig, vm *virtualMachine, sshRunner *crcssh.Runner) error {
-	if startConfig.Preset == crcPreset.OpenShift {
-		sizeToMove := ocpGetPVShiftSize(startConfig.DiskSize, startConfig.PersistentVolumeSize)
-		_, _, err := sshRunner.RunPrivileged("move topolvm partition to end of disk", fmt.Sprintf("echo '+%dG,' | sudo sfdisk --move-data /dev/vda -N 5 --force", sizeToMove))
-		var exitErr *ssh.ExitError
-		if err != nil {
-			if !errors.As(err, &exitErr) {
-				return err
-			}
-			if exitErr.ExitStatus() != 1 {
-				return err
-			}
-		}
-		if err == nil {
-			logging.Info("Restart VM after moving topolvm partition to end")
-			if err := restartHost(ctx, vm, sshRunner); err != nil {
-				return fmt.Errorf("Failed to move topolvm partition to increase root partition size: %w", err)
-			}
-		}
-	}
-	rootPart, err := getrootPartition(sshRunner, startConfig.Preset)
-	if err != nil {
-		return err
-	}
-
-	if err := growPersistentVolume(sshRunner, startConfig.Preset, startConfig.PersistentVolumeSize); err != nil {
-		return fmt.Errorf("Unable to grow persistent volume partition: %w", err)
-	}
-
-	// with '/dev/[sv]da4' as input, run 'growpart /dev/[sv]da 4'
-	if _, _, err := sshRunner.RunPrivileged(fmt.Sprintf("Growing %s partition", rootPart), "/usr/bin/growpart", rootPart[:len("/dev/.da")], rootPart[len("/dev/.da"):]); err != nil {
```

Comment:
The corresponding code in the new file has become
```
	if err := growPersistentVolume(sshRunner, startConfig.Preset, startConfig.PersistentVolumeSize); err != nil {
		return fmt.Errorf("Unable to grow persistent volume partition: %w", err)
	}

	// with '/dev/[sv]da4' as input, run 'growpart /dev/[sv]da 4'
	if err := growPartition(sshRunner, rootPart); err != nil {
		return nil
	}
```

While the addition of a helper is a good thing, this should not be hidden in a commit saying it's only moving code.
Fwiw, a helper for `move topolvm partition to end of disk` could also be useful to make `growRootFilesystem` easier to read.